packages/rum-core/src/domain/rumEventsCollection/action/trackClickActions.ts

```diff
@@ -221,59 +201,59 @@ function newClick(
   }
   const historyEntry = history.add(id, base.startClocks.relative)
   const eventCountsSubscription = trackEventCounts(lifeCycle)
-  let state: ClickState = { status: ClickStatus.ONGOING }
-  const frustrations = new Set<FrustrationType>()
+  let status = ClickStatus.ONGOING
+  let activityEndTime: undefined | TimeStamp
+  const frustrationTypes: FrustrationType[] = []
   const stopObservable = new Observable<void>()
 
-  function stop(endTime?: TimeStamp) {
-    if (state.status !== ClickStatus.ONGOING) {
+  function stop(newActivityEndTime?: TimeStamp) {
```

Comment:
Hum right. I'd say that when the view ends, then we can also consider that activity occured, since it will have a `loading_time`, and we shouldn't consider it as `dead`. WDYT?
Maybe adding a test about it would clarify the expected behavior?

Reply:
Ok so I tried to move it to `waitPageActivityEnd` (behind an option, because we don't want to do that for `view.loadingTime`, right?), but it's a bit annoying: to keep backward compatibility, we want to discard the Click if we don't track frustration, so we would need to convey this information in the `PageActivityEndEvent`. I think that it complicate things a bit.